packages/util/src/u8a/toBigInt.ts

```diff
@@ -23,7 +23,7 @@ export function u8aToBigInt (value: Uint8Array, { isLe = true, isNegative = fals
     : value.slice().reverse();
   const count = u8a.length;
 
-  if (isNegative && (!count || (u8a[count - 1] & 0x80))) {
+  if (isNegative && count && (u8a[count - 1] & 0x80)) {
```

Comment:
So to understand this, `isNegative` will not indicate if we want to convert a `signed` value, correct?
If so maybe it would be good to refactor the name to `isSigned` and deprecate `isNegative` option?

Reply:
The naming `isNegative` is indeed very unfortunate, it does mean `isSigned` and should have been named such 5 year ago. Got it right eventually [in the api](https://github.com/polkadot-js/api/blob/52e36534cfde94d4f032e88fee9ba0d24ccb4fb1/packages/types-codec/src/abstract/Int.ts#L97).

The timelines for deprecation (at least 6 months and then in the next major) and then the additional clutches it brings code & maintenance wise to support old/new has newer seemed to be worth the effort. So it is a wart.